modules/core/02-client/types/msgs.go

```diff
@@ -283,3 +284,39 @@ func (msg *MsgUpdateParams) ValidateBasic() error {
 	}
 	return msg.Params.Validate()
 }
+
+// NewMsgIBCSoftwareUpgrade creates a new MsgIBCSoftwareUpgrade instance
+func NewMsgIBCSoftwareUpgrade(authority string, plan upgradetypes.Plan, upgradedClientState exported.ClientState) (*MsgIBCSoftwareUpgrade, error) {
```

Comment:
can also add the typical interface conformance assertion at top of file, i.e `_ sdk.Msg = (*MsgIBCSoftwareUpgrade)(nil)`